UI/src/Hierarchy.cpp

```diff
@@ -103,74 +91,75 @@ void drawHierarchyLine(GameObject* gameObject) {
         }
     }
 
     // Draw children recursively if the node is open
     if (nodeOpen) {
-        for (auto* child : gameObject->getChildren()) {
+        for (auto* child : entity->getChildren()) {
             drawHierarchyLine(child);  // Use .get() to pass the raw pointer to drawHierarchyLine
         }
         ImGui::TreePop();
     }
 }
 
 // Function to draw the entire hierarchy window
-void drawHierarchy() {
+void drawHierarchy(Scene* s) {
     ImGui::Begin("Hierarchy");
 
-    // Create a new GameObject on right-click
-    if (ImGui::BeginPopupContextWindow()) {
-        if (ImGui::MenuItem("New GameObject")) {
-            std::string rootName = getNextRootObjectName();
-            GameObject* newObject = new GameObject(rootName); // Default name
-            gameObjects.emplace_back(newObject);
-            selectedGameObject = newObject; // Select newly created object
-            consoleLog("Created New GameObject: " + newObject->getName() + " with ID: " + std::to_string(reinterpret_cast<uintptr_t>(newObject)));
-            
-        }
+    Entity* rootEntity = s->getRoot();
 
-        ImGui::EndPopup();
+    if (rootEntity) {
+       drawHierarchyLine(rootEntity);
     }
 
-    // Loop over all root game objects in the scene
-    for (auto* rootObject : gameObjects) {
-        drawHierarchyLine(rootObject);
-    }
+    // Create a new GameObject on right-click
+    //if (ImGui::BeginPopupContextWindow()) {
+    //    if (ImGui::MenuItem("New GameObject")) {
+    //        std::string rootName = getNextRootObjectName();
+    //        GameObject* newObject = new GameObject(rootName); // Default name
+    //        gameObjects.emplace_back(newObject);
+    //        selectedGameObject = newObject; // Select newly created object
+    //        consoleLog("Created New GameObject: " + newObject->getName() + " with ID: " + std::to_string(reinterpret_cast<uintptr_t>(newObject)));
+    //        
+    //    }
+
+    //    ImGui::EndPopup();
+    //}
 
     ImGui::End();
 }
 
 // Function to retrieve the currently selected GameObject
-GameObject* getSelectedGameObject() {
-    return selectedGameObject;
+Entity* getSelectedEntity() {
+    return selectedEntity;
 }
 
 
-std::string getNextRootObjectName() {
-    int count = 0;
-    std::string baseName = "GameObject";
-    std::string newName = baseName;
-
-    // Count how many root objects have the name "GameObject", "GameObject(1)", etc.
-    for (const auto* rootObject : gameObjects) {
-        const std::string& rootName = rootObject->getName();
-        if (rootName == baseName || (rootName.find(baseName + "(") == 0 && rootName.back() == ')')) {
-            count++;
-        }
-    }
-
-    // Append the number if count > 0 (e.g., "GameObject(1)", "GameObject(2)")
-    if (count > 0) {
-        newName = baseName + "(" + std::to_string(count) + ")";
-    }
-
-    return newName;
-}
+//std::string getNextRootObjectName() {
```

Comment:
more commented out stuff, better to just delete. poWor oWold code but its for Great Good ~~!!